src/R/Core/Impl/AST/Operators/Associativity.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Core.AST.Operators {
+    // https://en.wikipedia.org/wiki/Operator_associativity
+    /// <summary>
+    /// Associativity. Consider the expression a~b~c. If the operator 
+    /// ~ has left associativity, this expression would be interpreted as (a~b)~c.
+    /// If the operator has right associativity, the expression would be
+    /// interpreted as a~(b~c).
+    /// </summary>
+    public enum Associativity {
```

Comment:
This is the reverse case - we treat something as valid R code syntactically, when it doesn't actually get parsed by R itself, so our design-time parsing is more strict than run-time parsing.

Anyway, it's just a random thing that came up while I was playing with operators. I suspect it's not trivial to fix, so I can file a separate issue.